packages/memory-inspector/src/browser/editable-widget/memory-editable-table-widget.tsx

```diff
@@ -0,0 +1,324 @@
+            return;
+        }
+        const arrayElementsPerRow = (this.options.byteSize / 8) * this.options.bytesPerGroup * this.options.groupsPerRow;
+        const isAlreadyEdited = this.pendingMemoryEdits.has(this.highlightedField.toString());
+        const oldValue = this.pendingMemoryEdits.get(initialHighlight.toString()) ??
+            this.memory.bytes[initialHighlightIndex.toInt()].toString(16).padStart(2, '0');
+        let possibleNewHighlight = new Long(-1);
+        let newValue = oldValue;
+        switch (keyCode) {
+            case Key.ARROW_DOWN.keyCode:
+                possibleNewHighlight = initialHighlight.add(arrayElementsPerRow);
+                event.preventDefault();
+                event.stopPropagation();
+                break;
+            case Key.ARROW_UP.keyCode:
+                possibleNewHighlight = initialHighlight.greaterThan(arrayElementsPerRow) ? initialHighlight.subtract(arrayElementsPerRow) : possibleNewHighlight;
+                event.preventDefault();
+                event.stopPropagation();
+                break;
+            case Key.ARROW_RIGHT.keyCode:
+                possibleNewHighlight = initialHighlight.add(1);
+                event.preventDefault();
+                event.stopPropagation();
+                break;
+            case Key.ARROW_LEFT.keyCode:
+                possibleNewHighlight = initialHighlight.greaterThan(0) ? initialHighlight.subtract(1) : possibleNewHighlight;
+                break;
+            case Key.BACKSPACE.keyCode:
+                newValue = oldValue.slice(0, oldValue.length - 1);
+                break;
+            case Key.DELETE.keyCode:
+                newValue = '';
+                break;
+            case Key.ENTER.keyCode:
+                this.submitMemoryEdits();
+                break;
+            case Key.ESCAPE.keyCode:
+                if (isAlreadyEdited) {
+                    this.clearEdits(this.highlightedField);
+                } else {
+                    this.clearEdits();
+                }
+                break;
+            default: {
+                const keyValue = parseInt(KeyCode.createKeyCode(event.nativeEvent).toString(), 16);
+                if (!Number.isNaN(keyValue)) {
+                    newValue = isAlreadyEdited ? oldValue : '';
+                    if (newValue.length < 2) {
+                        newValue += keyValue.toString(16);
+                    }
+                }
+            }
+        }
+        if (this.isInBounds(possibleNewHighlight)) {
+            this.highlightedField = possibleNewHighlight;
+        }
+        const valueWasChanged = newValue !== oldValue;
+        if (valueWasChanged) {
+            this.pendingMemoryEdits.set(this.highlightedField.toString(), newValue);
+        }
+        if (valueWasChanged || !this.highlightedField.equals(initialHighlight)) {
+            this.update();
+        }
+    };
+
+    private isInBounds(candidateAddress: Long): boolean {
```

Comment:
Could you please make this method protected?

Reply:
Done.